### ISSUE TITLE
Passing `*something` to a keyword arguments only function doesn't fail

### ISSUE THREAD

**Issue by patrick91** [microsoft/pyright#3200]
Looks like when defining a function that only accepts keyword arguments and passing `*something` to it doesn't fail.

Here's an example that should fail but passes type checking:

```python
a = {"f": "an example"}

def x(*, f: str = ''):
    print(f)

x(**a)
x(*a)  # this is invalid and should fail
```

output from latest pyright:

```
@patrick91 ➜ /workspaces/strawberry $ pyright demo.py 
No configuration file found.
pyproject.toml file found at /workspaces/strawberry.
Loading pyproject.toml file at /workspaces/strawberry/pyproject.toml
Assuming Python platform Linux
Searching for source files
Found 1 source file
0 errors, 0 warnings, 0 informations 
Completed in 0.724sec
```

and output from python:

```
an example
Traceback (most recent call last):
  File "demo.py", line 8, in <module>
    x(*a)
TypeError: x() takes 0 positional arguments but 1 was given
```

**Comment by erictraut** [comment]
It's not clear that this should generate an error because it might be a false positive. If `something` is a zero-length Iterable, this will not generate an error at runtime. Mypy doesn't generate an error in this case either.

**Comment by patrick91** [comment]
> It's not clear that this should generate an error because it might be a false positive. If `something` is a zero-length Iterable, this will not generate an error at runtime. 

is there any use case for doing something like this?

```python

def x(*, f: str = ''):
    print(f)

a = []

x(*a)
```

even if the list would be empty the code would fail as soon as we add anything to the list, since we can't use this function with positional arguments.

Maybe I'm missing some other use case?

> Mypy doesn't generate an error in this case either.
Yes, I tested mypy too and it didn't fail, but it still feels like a potential bug :)

**Comment by JelleZijlstra** [comment]
For what it's worth I did make this case an error in pyanalyze (and also for the similar case of sending `**kwargs` to a function with only positional-only parameters). On our internal codebase this check found a few case where a wrapper function was doing `def f(*args, **kwargs): inner(*args, **kwargs)` even if `inner` didn't accept `*args`. Pyright of course gets run on a lot more code than pyanalyze, though, and maybe there are other patterns where this check would lead to false positives.

**Comment by erictraut** [comment]
I think there are legitimate situations where the type of x is a union of multiple callable types.

Our general principle is that false negatives are preferable to false positives especially for general type checks like those that involve argument/parameter matching. I suspect that's the same philosophy that led mypy to not flag this as an error.

I'm going to close this for now, but we might reconsider adding it in the future if we get sufficient signal and we convince ourselves that it won't produce too many false positives.
